src/csharp/Grpc.Core/AsyncClientStreamingCall.cs

```diff
@@ -36,7 +36,21 @@ public sealed class AsyncClientStreamingCall<TRequest, TResponse> : IDisposable
         readonly Func<Metadata> getTrailersFunc;
         readonly Action disposeAction;
 
-        internal AsyncClientStreamingCall(IClientStreamWriter<TRequest> requestStream, Task<TResponse> responseAsync, Task<Metadata> responseHeadersAsync, Func<Status> getStatusFunc, Func<Metadata> getTrailersFunc, Action disposeAction)
+        /// <summary>
+        /// Creates a new AsyncClientStreamingCall object with the specified properties.
+        /// </summary>
+        /// <param name="requestStream">Stream of request values.</param>
+        /// <param name="responseAsync">The response of the asynchronous call.</param>
+        /// <param name="responseHeadersAsync">Response headers of the asynchronous call.</param>
+        /// <param name="getStatusFunc">Delegate returning the status of the call.</param>
+        /// <param name="getTrailersFunc">Delegate returning the trailing metadata of the call.</param>
+        /// <param name="disposeAction">Delegate to invoke when Dispose is called on the call object.</param>
+        public AsyncClientStreamingCall(IClientStreamWriter<TRequest> requestStream,
```

Comment:
alternatively, we can provide factory methods:
currently, we have 2 factories: https://github.com/grpc/grpc/blob/master/src/csharp/Grpc.Core/Calls.cs (for "actual" calls)
and https://github.com/grpc/grpc/blob/master/src/csharp/Grpc.Core.Testing/TestCalls.cs (for test doubles).
Perhaps we could have a 3rd factory class for "intercepted" calls. That way, we would have an intermediate layer in case we wanted to make changes to Async**Call constructors.

Reply:
The `ChannelExtensions` in the interceptor PR acts as the third factory, but I picked the extension route to make the API nicer, i.e. enable `channel.Intercept(...)`.